Jenkinsfile

```diff
@@ -1,42 +1,42 @@
-try {
-    node('slave') {
-        stage('check-out-code') {
-            dir('src') {
-                checkout scm
-            }
-        }
-
-        stage('test') {
-            dir('src') {
-                sh 'make test'
-            }
-        }
-
-        if (env.BRANCH_NAME == 'master') {
-            stage('deploy-to-prod') {
-                build job: 'puppet-trigger', parameters: [
-                    [$class: 'StringParameterValue', name: 'server', value: 'ns'],
-                ]
-            }
+pipeline {
+  agent {
+    label 'slave'
+  }
+  options {
+    ansiColor('xterm')
+    timeout(time: 1, unit: 'HOURS')
+  }
+  stages {
+    stage('check-out-code') {
+      steps {
+        checkout scm
+      }
+    }
+    stage('test') {
+      steps {
+        sh 'make test'
+      }
+    }
+    stage('deploy') {
+      when {
+        branch 'master'
+      }
+      steps {
+        agent {
+          label 'deploy'
         }
+        puppetTrigger('ns')
+      }
     }
-
-} catch (err) {
-    def subject = "${env.JOB_NAME} - Build #${env.BUILD_NUMBER} - Failure!"
-    def message = "${env.JOB_NAME} (#${env.BUILD_NUMBER}) failed: ${env.BUILD_URL}"
-
-    if (env.BRANCH_NAME == 'master') {
-        slackSend color: '#FF0000', message: message
-        mail to: 'root@ocf.berkeley.edu', subject: subject, body: message
-    } else {
-        mail to: emailextrecipients([
-            [$class: 'CulpritsRecipientProvider'],
-            [$class: 'DevelopersRecipientProvider']
-        ]), subject: subject, body: message
+  }
+  post {
+    failure {
+      emailNotification()
     }
-
-    throw err
+    always {
+      ircNotification()
```

Comment:
does this go to #rebuild? or to #rebuild-spam?

Reply:
[By default, `#rebuild-spam`](https://github.com/ocf/shared-pipeline/blob/a23a33c3cc7421146a14741fc0cc506cb28db7eb/vars/ircNotification.groovy#L7), but it's configurable by passing arguments to this function.